### PR TITLE
feat: add Views Management System for Filters

### PR DESCRIPTION
**Pull request by andrewgolovanov**
**Description:**
This pull request implements a new Views system that allows users to save, manage, and reuse filter configurations. This feature enables users to persist their filter setups and switch between different saved views.

**Key Features**
- Save current filter configuration as a named view
- Manage saved views (rename, reorder, delete)
- Update existing views when filter configuration changes
- Automatic validation of view names (prevents duplicates)
- Drag-and-drop reordering of saved views
- Local storage persistence for saved views

**Technical Implementation**
- Integrated with existing filter system using `FiltersBar` component
- Added validation system for view names and filter configurations
- Implemented drag-and-drop functionality using dnd-kit
- Added local storage integration for persistence

https://github.com/user-attachments/assets/45cfe1c3-fb19-47b4-affe-28dffe303ea8

### CODE REVIEW

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]
I think there's some mixup between two different features happening here.
In my understanding, the "View" dropdown was meant to switch between different layouts of the same content.. eg. list vs grid vs card layouts.
Vs. what you've implemented is the "Saved Filters" feature. This allows users to apply a few filters on the content, and save that for quick access later.

File: packages/ui/src/components/filters/filters-bar/actions/views.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/components/filters/filters-bar/actions/views.tsx:97]: same thing for all callbacks.. there should be no business logic in the `ui` package components and views. these react components should be presentation-only, and take callbacks as props for all behaviour

File: packages/ui/src/components/filters/filters-bar/actions/views.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/components/filters/filters-bar/actions/views.tsx:165]: can we externalise all display strings to appropriate language files in the first cut itself? it'll reduce re-work to add i18n later.

File: packages/ui/src/components/filters/manage-views.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/components/filters/manage-views.tsx:14]: when did we add `dnd-kit` as a dependency? 
we need to evaluate all new libraries and get app-sec approvals before we start using them.

Comment (abhinavrastogi-harness) [reply packages/ui/src/components/filters/manage-views.tsx:14]: also, cc @srdjan-harness because we are using some other dnd library for the pipeline studio

File: packages/ui/src/components/filters/manage-views.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/components/filters/manage-views.tsx:142]: i'd like to avoid non-english comments in the codebase. open to opinions though.

File: packages/ui/src/components/filters/manage-views.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/components/filters/manage-views.tsx:100]: this contains a LOT of business logic, which we decided to avoid in `ui` package.

we want to follow the MVVM pattern, where the "view" layer, should only be the presentational layer.

we want to be able to re-use the views across two different backends, which is not possible if we hard-code business logic in the view layer.

File: packages/ui/src/views/webhooks/webhook-list/repo-webhook-list-page.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/webhooks/webhook-list/repo-webhook-list-page.tsx:14]: same issue with `date-fns` package. this should not be added and neither should it be required.
I know this is existing code, but just making a note here for later.

File: packages/ui/src/views/webhooks/webhook-list/repo-webhook-list-page.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/webhooks/webhook-list/repo-webhook-list-page.tsx:28]: I don't think we need to pass `t` function down to all components. We can just call the `useTranslationStore` hook everywhere directly.

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]
In general I'm seeing a lot of mixed business logic in the shared views in this PR. Maybe let's discuss tomorrow before we merge this change.

File: packages/ui/src/components/button.tsx
Comment (andrewgolovanov) [reply packages/ui/src/components/button.tsx:29]: @abhinavrastogi-harness I only removed the vertical padding since it doesn’t make sense, as the button’s height is already defined using `h-`
image

File: packages/ui/locales/en/component_old.json
Comment (andrewgolovanov) [reply packages/ui/locales/en/component_old.json:1]: Unfortunately, it appeared on its own after the project was built.

File: packages/ui/src/components/filters/filters-bar/actions/views.tsx
Comment (andrewgolovanov) [reply packages/ui/src/components/filters/filters-bar/actions/views.tsx:106]: @abhinavrastogi-harness I’ve revised the approaches, so now only the local logic necessary for this component remains. It handles its own state, while all the business logic for filtering and validation has been moved to a hook called [use-view-management](https://github.com/harness/canary/pull/513/files#diff-c58401d04ae9a913353d34aeabc42330af54d16582e87e43325c039034e9c35bR43), located at `views/repo`.

File: packages/ui/src/components/filters/manage-views.tsx
Comment (andrewgolovanov) [reply packages/ui/src/components/filters/manage-views.tsx:16]: At the moment, I’ve left comments in the code to revisit this in a future PR.

File: packages/ui/src/components/filters/manage-views.tsx
Comment (andrewgolovanov) [reply packages/ui/src/components/filters/manage-views.tsx]: Here, all the business logic has also been moved to the [use-view-management](https://github.com/harness/canary/pull/513/files#diff-c58401d04ae9a913353d34aeabc42330af54d16582e87e43325c039034e9c35bR43) hook at the page level, which is used for validating and managing all the local states of this component.

**Review by 3em — APPROVED** [review]

File: packages/ui/src/components/filters/filters-bar/actions/views.tsx
Comment (andrewgolovanov) [reply packages/ui/src/components/filters/filters-bar/actions/views.tsx:198]: I’m planning to address this in the next PR so that this one doesn’t block further development.

File: packages/ui/src/views/webhooks/webhook-list/repo-webhook-list-page.tsx
Comment (andrewgolovanov) [reply packages/ui/src/views/webhooks/webhook-list/repo-webhook-list-page.tsx]: This will be fixed in the next PR

File: packages/ui/src/views/webhooks/webhook-list/repo-webhook-list-page.tsx
Comment (andrewgolovanov) [reply packages/ui/src/views/webhooks/webhook-list/repo-webhook-list-page.tsx]: I’ve left a comment in the code to revisit the approach to using `date-fns` in the future and to plan for fully phasing out its usage.

File: packages/ui/locales/en/component_old.json
Comment (abhinavrastogi-harness) [inline packages/ui/locales/en/component_old.json:1]: can we delete these `component_old` files? these should NOT be needed.

File: packages/ui/src/components/filters/filters-bar/actions/views.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/components/filters/filters-bar/actions/views.tsx:50]: this validation should not exist in reusable components

File: packages/ui/src/components/filters/filters-bar/actions/views.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/components/filters/filters-bar/actions/views.tsx:93]: this validation should not exist in reusable components

File: packages/ui/src/views/repo/hooks/use-view-management.ts
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/repo/hooks/use-view-management.ts:43]: this hook implements logic that seems specific to harness open-source, and should be implemented in the gitness app instead.

Comment (abhinavrastogi-harness) [reply packages/ui/src/views/repo/hooks/use-view-management.ts:43]: this should also be implemented using zustand stores once it's in gitness

File: packages/ui/src/views/repo/hooks/use-view-management.ts
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/repo/hooks/use-view-management.ts:1]: is this system multi-tenant? I mean, will this scale to manage views for multiple pages and resource types? eg. PR list, repo list, commit list, pipeline list etc.

**Review by abhinavrastogi-harness — APPROVED** [review]
unblocking this PR to merge, but I think we need to revisit filters later, at which point there will need to be a major refactor as we move all filtering logic to the backend.

File: packages/ui/src/views/repo/hooks/use-view-management.ts
Comment (andrewgolovanov) [reply packages/ui/src/views/repo/hooks/use-view-management.ts:1]: @abhinavrastogi-harness Yes, the current system is easily scalable and allows for unique filtering to be applied to each page.

**Pull request merged**
